### ISSUE TITLE
Escaped character in bash regex group => SC1072 (<- SC1073 <- SC1009)

### ISSUE THREAD

**Issue by Chessax** [koalaman/shellcheck#980]
- Rule Id: SC1072 (<- SC1073 <- SC1009)
- Version: Online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit.
- [x] I searched through [issues](https://github.com/koalaman/shellcheck/issues) and didn't find anything directly related.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

var="a.b"
if [[ $var =~ .(\.). ]]; then # Can be any escaped character
  echo "${BASH_REMATCH[1]}"
fi
```

#### Here's what shellcheck currently says:
``` 
Line 4:
if [[ "a.b" =~ .(\.). ]]; then
               ^-- SC1009: The mentioned parser error was in this regex.
                ^-- SC1073: Couldn't parse this regex grouping.
                  ^-- SC1072: Unexpected . Fix any mentioned problems and try again.

```

#### Here's what I wanted or expected to see:
```
No issues detected!
```

#### Here's what Bash 3.2.53 says (and I expected to see):
```
.
```

**Comment by omarkhan** [comment]
I only get this when I run the shellcheck docker container, not when I run it on the host machine. Strange.

**Comment by imrehg** [comment]
Running into the same issue on shellcheck version: 0.4.7, running on a normal machine (not in docker)

**Comment by e-kwsm** [comment]
Now (0.11.0) no issues are detected, so it seems to be fixed at some point.
